File: generate-service-images.py
from PIL import Image, ImageDraw, ImageFont
import os

# Create images directory if it doesn't exist
output_dir = r"c:\Users\Anuj Singh\Desktop\AI DigiNext\Website - Vibe Coding\AIpath Studio\public\images\services"
os.makedirs(output_dir, exist_ok=True)

# Service configurations with gradients
services = [
    {
        "name": "ai-chatbot",
        "icon": "💬",
        "gradient": [(59, 130, 246), (6, 182, 212)],  # blue to cyan
        "title": "AI Chatbots"
    },
    {
        "name": "automation",
        "icon": "⚡",
        "gradient": [(168, 85, 247), (236, 72, 153)],  # purple to pink
        "title": "Automations"
    },
    {
        "name": "branding",
        "icon": "🎨",
        "gradient": [(249, 115, 22), (239, 68, 68)],  # orange to red
        "title": "Product Branding"
    },
    {
        "name": "content-writing",
        "icon": "✍️",
        "gradient": [(34, 197, 94), (20, 184, 166)],  # green to teal
        "title": "Content Writing"
    },
    {
        "name": "social-media",
        "icon": "📱",
        "gradient": [(99, 102, 241), (168, 85, 247)],  # indigo to purple
        "title": "Social Media"
    },
    {
        "name": "ai-photoshoot",
        "icon": "📸",
        "gradient": [(236, 72, 153), (244, 63, 94)],  # pink to rose
        "title": "AI Photoshoots"
    },
    {
        "name": "ai-blogs",
        "icon": "📝",
        "gradient": [(234, 179, 8), (249, 115, 22)],  # yellow to orange
        "title": "AI-Based Blogs"
    },
    {
        "name": "email-campaigns",
        "icon": "📧",
        "gradient": [(59, 130, 246), (99, 102, 241)],  # blue to indigo
        "title": "Email Campaigns"
    },
    {
        "name": "lead-generation",
        "icon": "📍",
        "gradient": [(239, 68, 68), (236, 72, 153)],  # red to pink
        "title": "Lead Generation"
    }
]

def create_gradient(width, height, color1, color2):
    """Create a gradient image"""
    base = Image.new('RGB', (width, height), color1)
    top = Image.new('RGB', (width, height), color2)
    mask = Image.new('L', (width, height))
    mask_data = []
    for y in range(height):
        for x in range(width):
            # Diagonal gradient
            mask_data.append(int(255 * ((x + y) / (width + height))))
    mask.putdata(mask_data)
    base.paste(top, (0, 0), mask)
    return base

def add_pattern(draw, width, height):
    """Add geometric pattern overlay"""
    # Add circles
    for i in range(0, width + 200, 200):
        for j in range(0, height + 200, 200):
            draw.ellipse([i-100, j-100, i+100, j+100], outline=(255, 255, 255, 30), width=2)
    
    # Add lines
    for i in range(0, width, 100):
        draw.line([(i, 0), (i + height, height)], fill=(255, 255, 255, 20), width=1)

# Generate images
for service in services:
    print(f"Generating image for {service['title']}...")
    
    # Create base gradient
    width, height = 1200, 800
    img = create_gradient(width, height, service['gradient'][0], service['gradient'][1])
    
    # Add overlay for depth
    overlay = Image.new('RGBA', (width, height), (0, 0, 0, 0))
    draw = ImageDraw.Draw(overlay)
    
    # Add pattern
    add_pattern(draw, width, height)
    
    # Add semi-transparent overlay
    overlay_layer = Image.new('RGBA', (width, height), (0, 0, 0, 40))
    img = Image.alpha_composite(img.convert('RGBA'), overlay_layer)
    img = Image.alpha_composite(img, overlay)
    
    # Try to add text with icon
    try:
        # Use a larger font size for the icon
        font_size = 200
        try:
            # Try to use a system font that supports emojis
            font = ImageFont.truetype("seguiemj.ttf", font_size)
        except:
            try:
                font = ImageFont.truetype("arial.ttf", font_size)
            except:
                font = ImageFont.load_default()
        
        # Draw icon in center
        text = service['icon']
        draw = ImageDraw.Draw(img)
        
        # Get text bounding box
        bbox = draw.textbbox((0, 0), text, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]
        
        # Center position
        x = (width - text_width) // 2
        y = (height - text_height) // 2
        
        # Draw text with shadow
        draw.text((x+5, y+5), text, font=font, fill=(0, 0, 0, 100))
        draw.text((x, y), text, font=font, fill=(255, 255, 255, 200))
        
    except Exception as e:
        print(f"Could not add text: {e}")
    
    # Save image
    output_path = os.path.join(output_dir, f"{service['name']}.jpg")
    img.convert('RGB').save(output_path, 'JPEG', quality=90, optimize=True)
    print(f"✓ Saved: {output_path}")

print("\n✅ All service images generated successfully!")

File: generate-service-images.js
const fs = require('fs');
const path = require('path');

// Create images directory
const outputDir = path.join(__dirname, 'public', 'images', 'services');
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

// Service configurations
const services = [
  { name: 'ai-chatbot', title: 'AI Chatbots', icon: '💬', colors: ['#3B82F6', '#06B6D4'] },
  { name: 'automation', title: 'Automations', icon: '⚡', colors: ['#A855F7', '#EC4899'] },
  { name: 'branding', title: 'Product Branding', icon: '🎨', colors: ['#F97316', '#EF4444'] },
  { name: 'content-writing', title: 'Content Writing', icon: '✍️', colors: ['#22C55E', '#14B8A6'] },
  { name: 'social-media', title: 'Social Media', icon: '📱', colors: ['#6366F1', '#A855F7'] },
  { name: 'ai-photoshoot', title: 'AI Photoshoots', icon: '📸', colors: ['#EC4899', '#F43F5E'] },
  { name: 'ai-blogs', title: 'AI-Based Blogs', icon: '📝', colors: ['#EAB308', '#F97316'] },
  { name: 'email-campaigns', title: 'Email Campaigns', icon: '📧', colors: ['#3B82F6', '#6366F1'] },
  { name: 'lead-generation', title: 'Lead Generation', icon: '📍', colors: ['#EF4444', '#EC4899'] }
];

// Generate SVG placeholder images
services.forEach(service => {
  const svg = `
    <svg width="1200" height="800" xmlns="http://www.w3.org/2000/svg">
      <defs>
        <linearGradient id="grad-${service.name}" x1="0%" y1="0%" x2="100%" y2="100%">
          <stop offset="0%" style="stop-color:${service.colors[0]};stop-opacity:1" />
          <stop offset="100%" style="stop-color:${service.colors[1]};stop-opacity:1" />
        </linearGradient>
        <pattern id="pattern-${service.name}" x="0" y="0" width="100" height="100" patternUnits="userSpaceOnUse">
          <circle cx="50" cy="50" r="40" fill="none" stroke="rgba(255,255,255,0.1)" stroke-width="2"/>
          <line x1="0" y1="0" x2="100" y2="100" stroke="rgba(255,255,255,0.05)" stroke-width="1"/>
        </pattern>
      </defs>
      
      <!-- Gradient Background -->
      <rect width="1200" height="800" fill="url(#grad-${service.name})"/>
      
      <!-- Pattern Overlay -->
      <rect width="1200" height="800" fill="url(#pattern-${service.name})"/>
      
      <!-- Dark Overlay -->
      <rect width="1200" height="800" fill="rgba(0,0,0,0.15)"/>
      
      <!-- Icon -->
      <text x="600" y="450" font-size="200" text-anchor="middle" fill="rgba(255,255,255,0.9)" font-family="Arial, sans-serif">
        ${service.icon}
      </text>
      
      <!-- Title -->
      <text x="600" y="650" font-size="48" font-weight="bold" text-anchor="middle" fill="rgba(255,255,255,0.95)" font-family="Arial, sans-serif">
        ${service.title}
      </text>
      
      <!-- Decorative circles -->
      <circle cx="150" cy="150" r="80" fill="none" stroke="rgba(255,255,255,0.2)" stroke-width="3"/>
      <circle cx="1050" cy="650" r="100" fill="none" stroke="rgba(255,255,255,0.2)" stroke-width="3"/>
      <circle cx="200" cy="700" r="60" fill="none" stroke="rgba(255,255,255,0.15)" stroke-width="2"/>
      <circle cx="1000" cy="150" r="70" fill="none" stroke="rgba(255,255,255,0.15)" stroke-width="2"/>
    </svg>
  `;
  
  const outputPath = path.join(outputDir, `${service.name}.svg`);
  fs.writeFileSync(outputPath, svg.trim());
  console.log(`✓ Generated: ${service.name}.svg`);
});

console.log('\n✅ All service images generated successfully!');
console.log(`📁 Location: ${outputDir}`);